src/query/dblink_scan.c

```diff
@@ -120,39 +112,101 @@ dblink_get_basic_utype (T_CCI_U_EXT_TYPE u_ext_type)
     }
 }
 
-static void
+static int
+dblink_make_cci_value (DB_VALUE * cci_value, T_CCI_U_TYPE utype, void *val, int prec, int len)
+{
+  int error;
+
+  switch (utype)
+    {
+    case CCI_U_TYPE_BIGINT:
+      error = db_make_bigint (cci_value, *(DB_BIGINT *) val);
+      break;
+    case CCI_U_TYPE_INT:
+      error = db_make_int (cci_value, *(int *) val);
+      break;
+    case CCI_U_TYPE_FLOAT:
+      error = db_make_float (cci_value, *(float *) val);
+      break;
+    case CCI_U_TYPE_DOUBLE:
+      error = db_make_double (cci_value, *(double *) val);
+      break;
+    case CCI_U_TYPE_MONETARY:
+      error = db_make_monetary (cci_value, db_get_currency_default (), *(double *) val);
+      break;
+    case CCI_U_TYPE_STRING:
+      error = db_make_varchar (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    case CCI_U_TYPE_VARNCHAR:
+      error = db_make_varnchar (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    case CCI_U_TYPE_CHAR:
+      error = db_make_char (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    case CCI_U_TYPE_NCHAR:
+      error = db_make_nchar (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    default:
+      assert (false);
+      break;
+    }
+
+  return error;
+}
+
+static int
 dblink_make_date_time (T_CCI_U_TYPE utype, DB_VALUE * value_p, T_CCI_DATE * date_time)
 {
   DB_TIME t_time;
   DB_DATE t_date;
   DB_DATETIME t_datetime;
   DB_TIMESTAMP t_timestamp;
+  int error;
```

Comment:
it should be initialized
```suggestion
  int error = NO_ERROR;
```

Reply:
`value_p` also need to be initialized to null
```
db_make_null (value_p);
```

Reply:
Ok, thanks